src/main/java/edu/cornell/kfs/ksr/document/validation/impl/SecurityGroupRule.java

```diff
@@ -0,0 +1,170 @@
+		if (tabs == null) {
+			return false;
+		}
+		else if (tabs.size() > 0) {
+			Iterator<SecurityGroupTab> it = tabs.iterator();
+			while (it.hasNext()) {
+				SecurityGroupTab tab = it.next();
+				if (tab.isActive()) {
+					return true;
+				}
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Validates the maint. doc before it is written to the db.
+	 * 
+	 * @see org.kuali.rice.kns.maintenance.rules.MaintenanceDocumentRuleBase#dataDictionaryValidate(org.kuali.rice.kns.document.MaintenanceDocument)
+	 */
+	@Override
+	protected boolean dataDictionaryValidate(MaintenanceDocument document) {
+		boolean success = true;
+		SecurityGroup securityGroup = (SecurityGroup) document.getDocumentDataObject();
+		if (!hasAtLeastOneActiveTab(securityGroup.getSecurityGroupTabs())) {
+			GlobalVariables.getMessageMap().putErrorForSectionId(KSRConstants.SECTION_SECURITY_TABS, KSRKeyConstants.ERROR_SECURITY_GROUP_TAB_MISSING, new String());
+			success = false;
+		}
+
+		// Check the Security Group Name against the DB
+		Map<String, Object> hashMap = new HashMap<String, Object>();
+		hashMap.put(KSRConstants.SECURITY_GROUP_NAME, securityGroup.getSecurityGroupName());
+
+		BusinessObjectService businessObjectService = SpringContext.getBean(BusinessObjectService.class);
+		SecurityGroup temp = (SecurityGroup) businessObjectService.findByPrimaryKey(SecurityGroup.class, hashMap);
+
+		if (temp != null) {
+			if (securityGroup.getSecurityGroupId() != null) {
+				if (temp.getSecurityGroupName().equals(securityGroup.getSecurityGroupName())
+						&& (!temp.getSecurityGroupId().equals(securityGroup.getSecurityGroupId()))) {
+					GlobalVariables.getMessageMap().putError(KSRConstants.KSR_DOCUMENT_MAINTANABLE
+							+ "." + KSRConstants.SECURITY_GROUP_NAME, KSRKeyConstants.ERROR_SECURITY_GROUP_NAME_UNIQUE);
+					success = false;
+				}
+			}
+			else {
+				GlobalVariables.getMessageMap().putError(KSRConstants.KSR_DOCUMENT_MAINTANABLE
+						+ "." + KSRConstants.SECURITY_GROUP_NAME, KSRKeyConstants.ERROR_SECURITY_GROUP_NAME_UNIQUE);
+				success = false;
+			}
+
+		}
+
+		success = validateSecurityGroupTabs(document);
+
+		return (super.dataDictionaryValidate(document) && success);
+	}
+
+	/**
+	 * Determines uniqueness of each SecurityGroupTab in the maint. doc.
+	 * 
+	 * @param document
+	 *            - the current maint. doc
+	 * @return true if there are no duplicate tab orders or names
+	 */
+	public boolean validateSecurityGroupTabs(MaintenanceDocument document) {
+		boolean successTabOrder = true;
+		boolean successTabName = true;
+
+		List<SecurityGroupTab> securityGroupTabs = ((SecurityGroup) document.getDocumentDataObject()).getSecurityGroupTabs();
+		if (securityGroupTabs != null) {
+			for (int i = 0; i < securityGroupTabs.size(); i++) {
+				SecurityGroupTab tab = securityGroupTabs.get(i);
+
+				if ((tab.getTabOrder() == null) && (tab.getTabName() == null)) {
+					securityGroupTabs.remove(i);
```

Comment:
Will do some testing to see if this has some unwanted effects.

Reply:
@cah292 I've done some testing but I have not found an issue. Is there a specific scenario that you think will cause issues? Thank you.

Reply:
If a tab gets added/removed or the document is not an "Edit" one, that could mess up the code below that tries to restore the old tab data from the "old" maintainable. However, if the tab name and tab order fields have a worst case of being empty strings instead of nulls, then the problematic code shouldn't have any effect. Also, the document should be enforcing tab name and tab order as required fields anyway.

Perhaps the problematic section was meant to prevent null strings in the validation farther below that checks for duplicate names or tab orders, though the duplication checks could easily be updated to be null-safe.

Reply:
@cah292 I think it would be good to have a separate user story for this issue as you suggested. Thank you!

Reply:
I've created KFSPTS-21963 for further follow-up on this.